### PR TITLE
Wiki modal popup

### LINKED ISSUE #4932: Pop up a modal menu with templates when starting a new wiki page
Going to https://publiclab.org/wiki/new opens a new blank wiki page, but it's a bit bare, and hard for people to know where to begin. Let's offer a Bootstrap modal with some template options:

screen shot 2019-03-06 at 3 51 27 pm

The actual content of the modal we could set in a Feature ([learn about this system here](https://github.com/publiclab/plots2/blob/master/doc/DATA_MODEL.md#features)) which means we don't have to put the HTML for this into the codebase - we'll store it in the database, and I can help with that. For now, we can put that code into a Gist: https://gist.github.com

But the code for the modal which will contain the templates will have to go in our codebase. 

Here's the template for the "new wiki page" page; but note that we use this for editing the wiki page too:

https://github.com/publiclab/plots2/blob/master/app/views/wiki/edit.html.erb

To distinguish, we might need to set a variable in the controller that we can use to turn the modal on: 

https://github.com/publiclab/plots2/blob/master/app/controllers/wiki_controller.rb#L121

### PR DESCRIPTION
**Pull request by Chrisyee22**
Fixes #4932  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x ] PR is descriptively titled 📑 and links the original issue above 🔗
* [x ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by Chrisyee22** [comment]
@publiclab/reviewers can you please have a look at this to make sure I am on the correct path. TY 😄

File: Gemfile.lock [hunk 1]
```diff
@@ -263,7 +263,7 @@ GEM
     omniauth-github (1.3.0)
       omniauth (~> 1.5)
       omniauth-oauth2 (>= 1.4.0, < 2.0)
-    omniauth-google-oauth2 (0.6.0)
+    omniauth-google-oauth2 (0.6.1)
       jwt (>= 2.0)
```
Comment (gauravano) [inline Gemfile.lock]: Please exclude changes done in this file.

File: app/views/wiki/edit.html.erb [hunk 2]
```diff
@@ -72,7 +72,7 @@
 
       <%= render partial: 'editor/editor' %>
 
-      <a id="publish" tabindex="5" class="publish btn btn-primary btn-lg"><%= t('wiki.edit.publish') %></a>
+      <a id="publish" tabindex="5" class="publish btn btn-primary btn-lg" ><%= t('wiki.edit.publish') %></a>
```
Comment (gauravano) [inline app/views/wiki/edit.html.erb]:
```suggestion
      <a id="publish" tabindex="5" class="publish btn btn-primary btn-lg"><%= t('wiki.edit.publish') %></a>
```

**Comment by gauravano** [comment]
Hey @Chrisyee22, I tried your PR locally but the modal didn't opened when I opened page for creating new wiki. Can you post a gif or screenshots? Thanks!

**Comment by jainaman224** [comment]
@Chrisyee22 Do you need any assistance?

**Comment by Chrisyee22** [comment]
@gauravano sorry about the slow response. I just saw these messages. I’ll send a gif later today. 🙂

**Comment by Chrisyee22** [comment]
Here's a link to a gif of the modal popup.
https://gph.is/g/EByX6G4

**Comment by jainaman224** [comment]
Hey @Chrisyee22 Please fix PR checkboxes using `[x]`.

**Comment by gauravano** [comment]
for sure, you're going right here @Chrisyee22. do you want help with further designing?

**Comment by Chrisyee22** [comment]
@gauravano some help would be great. I don’t know what you all want for images as well as links. I was hoping that would work as a template that could be expanded upon.

**Comment by jywarren** [comment]
Hi, all! This is looking good - it would have to load immediately upon
going to https://publiclab.org/wiki, even earlier than shown in the gif --
does that make sense?

Also, we can use a Feature for the content of the modal, so you don't have
to worry too much about that part! Read more here:

https://github.com/publiclab/plots2/blob/master/doc/DATA_MODEL.md#features

Thank you!!!

**Comment by gauravano** [comment]
Ok. So, as @jywarren said, you don't have to worry about the modal content. So, here's updated checklist @Chrisyee22

1. Ensure that modal opens immediately after page opening
2.  Put the below code in the modal:

```
<% cache('feature_wiki-modal', skip_digest: true) do %>
    <%= feature('wiki-modal') %>
<% end %>
```
This will fetch the modal code from the view. 

3. Design the modal feature. You can do that or we can ask @publiclab/plots2-reviewers as we have many designers in team now :smile: 

Thanks!

**Comment by SidharthBansal** [comment]
Hi, do you require any help @Chrisyee22 ?
There are some conflicts. Can you please resolve them whenever you get some time?

**Comment by Chrisyee22** [comment]
@SidharthBansal Yeah I will have some time early next week if that's ok. 😄

**Comment by jywarren** [comment]
Hi, just checking if you've gotten stuck on this at all, or if I could help in any way? Thanks!

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR OR reopen this PR and add additional changes to it.
Thanks for contributing on Public Lab

**Pull request closed without merging**